src/figuresClasses.ts

```diff
@@ -1,19 +1,74 @@
+
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape,
+  color: Color,
 
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const [shortSide1, shortSide2, longSide] = [a, b, c].sort((f, s) => f - s);
+
+    if (a <= 0 || b <= 0 || c <= 0 || longSide >= (shortSide1 + shortSide2)) {
+      throw new Error('Incorrect side length');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const area = Math.floor((Math.sqrt(p * (p - this.a)
+    * (p - this.b) * (p - this.c))) * 100) / 100;
```

Comment:
Think of a ways of making your formatting to help understand the code
```suggestion
    const area = Math.floor((
      Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c)
    )) * 100) / 100;
```